### PR TITLE
[fix](variant) use lock to protect schema from threading unsafe

### PR DESCRIPTION
**Pull request by eldenmoon**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: be/src/olap/base_tablet.h [hunk 1]
```diff
@@ -279,7 +279,10 @@ class BaseTablet {
                                         int limit);
 
     // Return the merged schema of all rowsets
-    virtual TabletSchemaSPtr merged_tablet_schema() const { return _max_version_schema; }
+    virtual TabletSchemaSPtr merged_tablet_schema() const {
+        std::shared_lock rlock(_meta_lock);
+        return _max_version_schema;
```
Comment (cambyzju) [inline be/src/olap/base_tablet.h:284]: 返回指针感觉就是原子的，需要加锁吗？

Comment (eldenmoon) [reply be/src/olap/base_tablet.h:284]: _max_version_schema 原有结构可能有线程修改并析构(赋值过程中)，这里 不加锁可能返回一个析构后的指针，造成heap use after free

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Pull request merged**